misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/service/nodevisitors/ActionNodeVisitor.java

```diff
@@ -154,11 +166,32 @@ public void visit(RemoteMethodCallActionNode remoteMethodCallActionNode) {
 
     @Override
     public void visit(FunctionCallExpressionNode functionCallExpressionNode) {
-
-        if (functionCallExpressionNode.functionName() instanceof SimpleNameReferenceNode) {
-            String methodName = ((SimpleNameReferenceNode) functionCallExpressionNode.functionName()).name().text();
+        if ((functionCallExpressionNode.functionName() instanceof SimpleNameReferenceNode) &&
+                visitedFunctionNames.stream().noneMatch(functionName ->
+                        (functionName instanceof SimpleNameReferenceNode) &&
+                        ((SimpleNameReferenceNode) functionName).name().text()
+                                .equals(((SimpleNameReferenceNode) functionCallExpressionNode.functionName())
+                                        .name().text()))) {
```

Comment:
Shall we move this check into a separate function, instead of using multilevel blocks within if conditions?